src/main/java/org/folio/rest/impl/PutOrdersByIdHelper.java

```diff
@@ -114,28 +166,31 @@ public void updateOrderWithPoLines(String id, CompositePurchaseOrder compPO) {
   }
 
   private List<CompletableFuture<?>> processPoLinesCreation(CompositePurchaseOrder compOrder, JsonArray poLinesFromStorage) {
-    List<CompletableFuture<?>> futures = new ArrayList<>();
     boolean updateInventory = compOrder.getWorkflowStatus() == CompositePurchaseOrder.WorkflowStatus.OPEN;
-    for (PoLine line : compOrder.getPoLines()) {
-      boolean isNew = true;
-      for (int i = 0; i < poLinesFromStorage.size(); i++) {
-        JsonObject lineFromStorage = poLinesFromStorage.getJsonObject(i);
-        if (StringUtils.equals(lineFromStorage.getString("id"), line.getId())) {
-          isNew = false;
-          break;
-        }
-      }
-      if (isNew) {
-        futures.add(postOrderLineHelper.createPoLine(line, updateInventory));
-      }
+    return  compOrder.getPoLines().stream().filter(poLine ->
+      poLinesFromStorage.stream()
+        .map(o -> ((JsonObject) o).getString(ID))
+        .noneMatch(s -> StringUtils.equals(s, poLine.getId()))
+    ).map(poLine -> postOrderLineHelper.createPoLine(poLine, updateInventory))
+      .collect(Collectors.toList());
+  }
+
+  private String buildNewPoLineNumber(JsonObject poLineFromStorage, String poNumber) {
+    String oldPoLineNumber = poLineFromStorage.getString(PO_LINE_NUMBER);
+    Matcher matcher = PO_LINE_NUMBER_PATTERN.matcher(oldPoLineNumber);
+    if (matcher.find()) {
+      return poNumber + matcher.group(2);
     }
-    return futures;
+    return oldPoLineNumber;
```

Comment:
This should never really happen, but if it does I'm not sure leaving the poLineNumber as-is (which doesn't match the new PO Number) is the right thing to do.  At the absolute bare minimum we want to log an error.  Ideally we'd simply assign the line a new, valid number using the poNumber and the next poLine sequence number (which isn't implemented yet).  So for now let's just log an error and add a TODO here so we don't lose track of this and can circle back once the POLine sequence/API is in place.